cmd/bridge/downloads/server.go

```diff
@@ -0,0 +1,258 @@
+package downloadsserver
+
+import (
+	"strings"
+	"net/http"
+	"fmt"
+	"os"
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"path/filepath"
+	"io/ioutil"
+)
+
+// credits to https://www.arthurkoziel.com/writing-tar-gz-files-in-go/
+func addToArchive(tw *tar.Writer, filename string) error {
+	// Open the file which will be written into the archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Get FileInfo about our file providing file size, mode, etc.
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// Create a tar Header from the FileInfo data
+	header, err := tar.FileInfoHeader(info, info.Name())
+	if err != nil {
+		return err
+	}
+
+	// Use full path as name (FileInfoHeader only takes the basename)
+	// If we don't do this the directory strucuture would
+	// not be preserved
+	// https://golang.org/src/archive/tar/common.go?#L626
+	header.Name = filename
+
+	// Write file header to the tar archive
+	err = tw.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+
+	// Copy file content to tar archive
+	_, err = io.Copy(tw, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createArchive(tarName string, path string) error {
+	tarFile, err := os.Create(tarName)
+	if err != nil {
+		fmt.Println("Error creating tar file:", err)
+		return err
+	}
+	defer tarFile.Close() 
+	// new writers for gzip and tar
+	gw := gzip.NewWriter(tarFile)
+	defer gw.Close()   
+	tw := tar.NewWriter(tarFile)
+	defer tw.Close()   
+
+	err = addToArchive(tw, path)
+	if err != nil {
+		return err
+	}
+	
+	return nil
+}
+ 	
+func writeIndex(path string, message string) error {
+	content := strings.Join([]string{
+		"<!doctype html>",
```

Comment:
Why do we need a HTML layout? The server should  be just serving the archives.

Reply:
The original server that this story aims to replace served these files in this format. Thus I assumed this is the expected format for the consumers of the server providing cli-artifacts so I kept it the same.